pkg/infrastructure/ibmcloud/clusterapi/clusterapi.go

```diff
@@ -369,11 +369,19 @@ func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]
 	}
 	logrus.Debugf("bootstrap ignition config upload complete to %s/%s/%s", cosInstanceName, bucketName, ignitionFile)
 
-	ignitionURL := url.URL{
-		Scheme: "https",
-		Host:   cosEndpoint,
-		Path:   fmt.Sprintf("%s/%s", bucketName, ignitionFile),
+	// Build the URL for the ignition config.
+	cosURL, err := url.Parse(cosEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse ibmcloud cos url: %s", cosURL)
+	}
```

Comment:
We should also print out the error here?

```suggestion
	if err != nil {
		return nil, fmt.Errorf("failed to parse ibmcloud cos url %s: %w", cosURL, err)
	}
```

Reply:
ah, oops, yeah.

Reply:
udpated